### ISSUE TITLE
What is difference between EveryDream and DreamBooth?

### ISSUE THREAD

**Issue by chesha1** [victorchall/EveryDream2trainer#23]
I may be wrong, but I look through docs but there's no details about the theory of EveryDream.
DreamBooth uses one prompt to describe the whole training dataset.
EveryDream uses a fully-labelled dataset creating by BLIP.
Is the above thought correct?

**Comment by victorchall** [comment]
EveryDream does not use these specific techniques that Dreambooth implementations do:

1.  Using generated images from the model itself for "preservation" purposes (to avoid forgetting and artifacts from your training images "bleeding" into the rest of the model).  I suggest you mix in "ground truth" data which would be ML image datasets, webscrapes from Laion or otherwise, etc.  Using generated images can reinforce the error already present in the model, like 6 fingers, 3 arms, 1 leg, etc.
2. "token" or "class" stuff is replaced by using captions per image.  There are no hard idea of training specific "concepts".  The concepts are whatever you put in your caption, and the caption is free form, not limited to specific identical words or phrases. You can mention everything in the image in your caption, and basically extract additional value by describing the entire scene instead of it just being trained as "bob smith" or "painting by XYZ artist".  You can caption like "bob smith in a brown jacket sitting on a bench in a park with trees in the background".

There's some paragraphs here in the old trainer repo that maybe help explain why I moved away from Dreambooth: https://github.com/victorchall/EveryDream-trainer#onward-to-every-dream

https://github.com/victorchall/EveryDream-trainer#ground-truth-data-sources-and-data-engineering-for-larger-scale-training

**Comment by chesha1** [comment]
> EveryDream does not use these specific techniques that Dreambooth implementations do:
> 
> 1. Using generated images from the model itself for "preservation" purposes (to avoid forgetting and artifacts from your training images "bleeding" into the rest of the model).  I suggest you mix in "ground truth" data which would be ML image datasets, webscrapes from Laion or otherwise, etc.  Using generated images can reinforce the error already present in the model, like 6 fingers, 3 arms, 1 leg, etc.
> 2. "token" or "class" stuff is replaced by using captions per image.  There are no hard idea of training specific "concepts".  The concepts are whatever you put in your caption, and the caption is free form, not limited to specific identical words or phrases. You can mention everything in the image in your caption, and basically extract additional value by describing the entire scene instead of it just being trained as "bob smith" or "painting by XYZ artist".  You can caption like "bob smith in a brown jacket sitting on a bench in a park with trees in the background".
> 
> There's some paragraphs here in the old trainer repo that maybe help explain why I moved away from Dreambooth: https://github.com/victorchall/EveryDream-trainer#onward-to-every-dream
> 
> https://github.com/victorchall/EveryDream-trainer#ground-truth-data-sources-and-data-engineering-for-larger-scale-training

Thanks for reply. I have another question:
Does EveryDream fine-tune the whole network expect for VAE like DreamBooth?
There are many other ways of fine-tuning, like textual inversion, hypernetworks and embeddings, which only add something to the original network or only change little parts of the network, which is theoretically have worse performance than Dreambooth.

**Comment by victorchall** [comment]
Both the text encoder andunet are trained by default. 

You can flip the text encoder and Unet training off, each with args, they are in advanced tweaking doc.  There has been some discussion that turning off text encoder training after some training may be useful to avoid it specifically being overtrained, so it is supported by using chaining and using disable_textenc_training.  Someone asked for disable_unet_training as well so I added it, but its not something I've really tested.  I'm not sure how useful that is.

My personal experience is turning off text encoder training may not be very valuable for training larger sets with detailed captions.  It may be useful for training smaller sets with simpler captions? 

Indeed, VAE is not tuned, but you can swap out the VAE on your own with autowebui or paste any diffusers VAE into your base model (replace ckpt_cache/SD21/vae/diffusion_pytorch_model.bin or something like that).  I may see if I can add VAE tuning at some point, but I feel that would likely be a separate process.  I believe VAEs for Stable Diffusion are more or less all interchangeable, so if you find a CKPT file with a VAE you want to try you can run the conversion script and copy out the VAE .bin file and put it in a model of your own in your /logs file, train for 1 epoch with 1 image and get a .CKPT file.

I only support fine tuning.  No textual inversion, HN, etc.  That spaces is very crowded and I don't know if I could add anything special to those processes, so I choose to devote my time to focus on "full" fine tuning and making that as robust and as powerful as possible.  Since this is just a hobby project my time is limited and adding other stuff would take away from time to focus on full fine tuning.

**Comment by chesha1** [comment]
> Both the text encoder andunet are trained by default.
> 
> You can flip the text encoder and Unet training off, each with args, they are in advanced tweaking doc. There has been some discussion that turning off text encoder training after some training may be useful to avoid it specifically being overtrained, so it is supported by using chaining and using disable_textenc_training. Someone asked for disable_unet_training as well so I added it, but its not something I've really tested. I'm not sure how useful that is.
> 
> My personal experience is turning off text encoder training may not be very valuable for training larger sets with detailed captions. It may be useful for training smaller sets with simpler captions?
> 
> Indeed, VAE is not tuned, but you can swap out the VAE on your own with autowebui or paste any diffusers VAE into your base model (replace ckpt_cache/SD21/vae/diffusion_pytorch_model.bin or something like that). I may see if I can add VAE tuning at some point, but I feel that would likely be a separate process. I believe VAEs for Stable Diffusion are more or less all interchangeable, so if you find a CKPT file with a VAE you want to try you can run the conversion script and copy out the VAE .bin file and put it in a model of your own in your /logs file, train for 1 epoch with 1 image and get a .CKPT file.
> 
> I only support fine tuning. No textual inversion, HN, etc. That spaces is very crowded and I don't know if I could add anything special to those processes, so I choose to devote my time to focus on "full" fine tuning and making that as robust and as powerful as possible. Since this is just a hobby project my time is limited and adding other stuff would take away from time to focus on full fine tuning.

Full fine tuning is definitely a greater idea.
Ways of treating dataset in EveryDream is much better than DreamBooth, the idea of prior preservation loss is so weird that most deep learning research do not do that way. Thanks for your work.

**Comment by crapthings** [comment]
How this compare to OneTrainer?
Are they same thing?
